src/librustc_mir/interpret/const_eval.rs

```diff
@@ -419,6 +459,42 @@ impl<'mir, 'tcx> super::Machine<'mir, 'tcx> for CompileTimeEvaluator {
     }
 }
 
+fn place_field<'a, 'tcx, 'mir>(
+    ecx: &mut EvalContext<'a, 'mir, 'tcx, CompileTimeEvaluator>,
+    i: usize,
+    place: Place,
+    layout: layout::TyLayout<'tcx>,
+) -> EvalResult<'tcx, Value> {
+    use rustc_data_structures::indexed_vec::Idx;
+    let (place, layout) = ecx.place_field(place, mir::Field::new(i), layout)?;
+    let val = ecx.read_place(place)?;
+    ecx.try_read_by_ref(val, layout.ty)
+}
+
+fn to_str<'a, 'tcx, 'mir>(
+    ecx: &mut EvalContext<'a, 'mir, 'tcx, CompileTimeEvaluator>,
+    val: Value,
+) -> EvalResult<'tcx, String> {
+    if let Value::ScalarPair(ptr, len) = val {
+        let len = len.to_bits(ecx.memory.pointer_size())?;
+        let bytes = ecx.memory.read_bytes(ptr, Size::from_bytes(len as u64))?;
+        let str = ::std::str::from_utf8(bytes).map_err(|err| EvalErrorKind::ValidationFailure(err.to_string()))?;
+        Ok(str.to_string())
```

Comment:
... and intern to `Symbol` here, without allocating a copy of the string slice!

Reply:
Oh, because the original value also came from a symbol? Or for making cloning the error cheap?